app/src/main/java/com/duckduckgo/app/privacy/ui/PrivacyDashboardViewModel.kt

```diff
@@ -62,32 +62,39 @@ class PrivacyDashboardViewModel(private val settingsStore: PrivacySettingsStore,
 
     init {
         resetViewState()
-        networkPercentsData.observeForever(networkPercentsObserver)
+        domainsVisited.observeForever(domainsVisitedObserver)
+        trackerNetworkTally.observeForever(trackerNetworkActivityObserver)
     }
 
     @VisibleForTesting
     public override fun onCleared() {
         super.onCleared()
-        networkPercentsData.removeObserver(networkPercentsObserver)
+        domainsVisited.removeObserver(domainsVisitedObserver)
+        trackerNetworkTally.removeObserver(trackerNetworkActivityObserver)
     }
 
-    fun onNetworkPercentsChanged(networkPercents: Array<NetworkPercent>) {
-
-        val enoughNetworksDetected = networkPercents.size >= 3
-        val enoughDomainsVisited = enoughNetworksDetected && networkPercents[0].totalDomainsVisited > 10
-        val showSummary = enoughDomainsVisited && enoughNetworksDetected
+    fun onDomainsVisitedChanged(count: Int?) {
+        val domainCount = count ?: 0
+        val networkCount = viewState.value?.trackerNetworkTally?.count() ?: 0
+        viewState.value = viewState.value?.copy(
+            showTrackerNetworkLeaderboard = showTrackerNetworkLeaderboard(domainCount, networkCount),
+            domainsVisited = domainCount
+        )
+    }
 
+    fun onTrackerNetworkTallyChanged(tally: List<NetworkTally>?) {
+        val domainCount = viewState.value?.domainsVisited ?: 0
+        val networkTally = tally ?: emptyList()
         viewState.value = viewState.value?.copy(
-                showNetworkTrackerSummary = enoughNetworksDetected && enoughDomainsVisited,
-                networkTrackerSummaryName1 = if (showSummary) networkPercents[0].networkName else null,
-                networkTrackerSummaryName2 = if (showSummary) networkPercents[1].networkName else null,
-                networkTrackerSummaryName3 = if (showSummary) networkPercents[2].networkName else null,
-                networkTrackerSummaryPercent1 = if (showSummary) networkPercents[0].percent else 0.0f,
-                networkTrackerSummaryPercent2 = if (showSummary) networkPercents[1].percent else 0.0f,
-                networkTrackerSummaryPercent3 = if (showSummary) networkPercents[2].percent else 0.0f
+            showTrackerNetworkLeaderboard = showTrackerNetworkLeaderboard(domainCount, networkTally.count()),
+            trackerNetworkTally = networkTally
         )
     }
 
+    private fun showTrackerNetworkLeaderboard (domainCount: Int, networkCount: Int): Boolean {
+        return domainCount >= LEADERNOARD_MIN_DOMAINS && networkCount >= LEADERBOARD_MIN_NETWORKS
```

Comment:
on iOS this is:

`sitesVisited() > visitedSitesThreshold && networksDetected().count >= 3`

Not sure which one is right tbh.

Reply:
Good point. It's not clear to me either. I have updated android for consistency.